client-side/src/stories/table/TableComponent.jsx

```diff
@@ -1,38 +1,166 @@
+
 import React from 'react';
 import { DataGrid } from '@mui/x-data-grid';
-import PropTypes from 'prop-types'
+import PropTypes from 'prop-types';
+import IconButton from '@mui/material/IconButton';
+import TextField from '@mui/material/TextField';
+import Select from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import Tooltip from '@mui/material/Tooltip';
+import GenericButton from '../Button/GenericButton.jsx';
 import './TableComponent.scss';
 
-const TableComponent = ({ dataObject, widthOfTable = "80%" ,widthOfColums}) => {
-  let columns = dataObject.headers.map((header,i) => ({
+const TableComponent = ({
+  dataObject,
+  widthOfTable,
+  widthOfColums,
+  actions,
+  editRowId,
+  handleFieldChange,
+  statusOptions,
+  addButton,
+  handleAddRow,
+  pageSize 
+}) => {
+  let columns = dataObject.headers.map((header, i) => ({
     field: header,
     headerName: header,
     width: widthOfColums[i],
-    align:'center',
-    headerAlign: 'center'
+    align: 'center',
+    headerAlign: 'center',
+    renderCell: (params) => {
+      if (editRowId && editRowId === params.row.id && header !== 'Actions') {
+        if (header === 'status') {
+          return (
+            <Select
+              value={params.value}
+              onChange={(e) => handleFieldChange && handleFieldChange(e, params.row.id)}
+              name={header}
+              fullWidth
+            >
+              {statusOptions.map(option => (
+                <MenuItem key={option.value} value={option.value}>
+                  {option.text}
+                </MenuItem>
+              ))}
+            </Select>
+          );
+        }
+        if (header === 'name') {
+          return (
+            <Tooltip title='This field is optional'>
+              <TextField
+                value={params.value}
+                onChange={(e) => handleFieldChange && handleFieldChange(e, params.row.id)}
+                name={header}
+                fullWidth
+              />
+            </Tooltip>
+          );
+        }
+        return (
+          <TextField
+            value={params.value}
+            onChange={(e) => handleFieldChange && handleFieldChange(e, params.row.id)}
+            name={header}
+            fullWidth
+          />
+        );
+      }
+      if (header === 'Actions') {
+        return (
+          <div>
+            {actions && actions.filter(action => action.condition(params.row.id)).map((action, index) => (
+              <Tooltip key={index} title={action.label}>
+                <IconButton onClick={() => action.func(params.row.id)} aria-label={action.label}>
+                  {React.createElement(action.icon)}
+                </IconButton>
+              </Tooltip>
+            ))}
+          </div>
+        );
+      }
+      return params.value;
+    },
   }));
 
+  if (!columns.some(column => column.field === 'Actions')) {
+    columns.push({
+      field: 'Actions',
+      headerName: 'Actions',
```

Comment:
what is the difference between this and what you have in line 70 - 83, are they the same?